apps/ui/src/contexts/index.tsx

```diff
@@ -23,10 +44,96 @@ export const GliaswapProvider: React.FC = (props) => {
   return (
     <QueryClientProvider client={queryClient}>
       <AdapterProvider adapter={adapter}>
-        <AssetProvider api={api} assetList={assetList}>
-          {props.children}
-        </AssetProvider>
+        {assetList.length > 0 ? (
+          <AssetProvider api={api} assetList={assetList}>
+            {props.children}
+          </AssetProvider>
+        ) : null}
       </AdapterProvider>
     </QueryClientProvider>
   );
 };
+
+interface GliaswapState {
+  /**
+   * the WalletConnectAdapter
+   * @see {AdapterContextState}
+   */
+  adapter: AdapterContextState<Web3ModalAdapter>;
+  // global assets info with balance
+  realtimeAssets: RealtimeInfo<GliaswapAssetWithBalance[]>;
+  /**
+   * an implement of GliaswapAPI
+   * @see {GliaswapAPI}
+   */
+  api: GliaswapAPI;
+  // wallet connect status
+  walletConnectStatus: ConnectStatus;
+  // when a wallet was connected, the user lock would be filled
+  currentUserLock?: Script;
+
+  currentCkbAddress: string;
+
+  currentEthAddress: string;
+}
+
+export function useGliaswap(): GliaswapState {
```

Comment:
the current useGliaswap is locale in `hooks/useGliaswap`, the `contexts` is not recommend using in components